proto/src/determined/resourcepool/v1/resourcepool.proto

```diff
@@ -0,0 +1,182 @@
+syntax = "proto3";
+
+package determined.resourcepool.v1;
+option go_package = "github.com/determined-ai/determined/proto/pkg/resourcepoolv1";
+
+
+
+// A Resource Pool is a pool of resources where containers are run.
+message ResourcePool {
```

Comment:
could you please also add annotations and mark required fields? https://github.com/determined-ai/determined/blob/cc491f3cde4faeb8f1e441f39a58f43300eb4a46/proto/src/determined/api/v1/trial.proto#L71

Reply:
same for any other proto messages you define

Reply:
maybe the user has to define some of these when setting up resource pools whereas some other fields can be left blank?